frontend/src/pages/ActivityReport/index.js

```diff
@@ -285,6 +314,7 @@ function ActivityReport({
         },
       );
 
+      updateFormData(updateGoals(savedReport));
```

Comment:
It appears to be passing the current state form data to updateGoals() along with the new form data.

Might be easier to read if you just pass both as individual params.